bundle/permissions/filter.go

```diff
@@ -0,0 +1,80 @@
+package permissions
+
+import (
+	"context"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/dyn"
+)
+
+type filterCurrentUser struct{}
+
+// The databricks terraform provider does not allow changing the permissions of
+// current user. The current user is implied to be the owner of all deployed resources.
+// This mutator removes the current user from the permissions of all resources.
+func FilterCurrentUser() bundle.Mutator {
+	return &filterCurrentUser{}
+}
+
+func (m *filterCurrentUser) Name() string {
+	return "FilterCurrentUserFromPermissions"
+}
+
+func filter(currentUser string) dyn.WalkValueFunc {
+	return func(p dyn.Path, v dyn.Value) (dyn.Value, error) {
+		// Permissions are defined at top level of a resource. We can skip walking
+		// after a depth of 4.
+		// [resource_type].[resource_name].[permissions].[array_index]
+		// Example: pipelines.foo.permissions.0
+		if len(p) > 4 {
```

Comment:
Do you really need this clause? The condition below will fire up earlier anyway if I understand correct and you won't go further than taht

Reply:
Yeah you are right, it'll be the same. If in the future the permissions block becomes more complex though this will act as another boundary condition. Being more conservative about the boundary conditions helps reason about it better IMO.

Reply:
It works both ways, additional conditions add additional cognitive load into what is happening. From what we're trying to achieve, the code might have looked like

```
if len(p) == 4 && len[2] == "permissions" {
// do mutation
}

return v, nil
```

if you inverse this it will look like 

```
 if len(p) != 4 || len[2] != "permissions" {
   return v, nil
 }
 
 // do mutate
```

It does not utilise ErrSkip though but makes it easier to reason about